be1-go/hub/witness.go

```diff
@@ -0,0 +1,77 @@
+package hub
+
+import (
+	"log"
+	"sync"
+
+	"go.dedis.ch/kyber/v3"
+)
+
+type witnessHub struct {
+	messageChan chan IncomingMessage
+
+	sync.RWMutex
+	channelByID map[string]Channel
+
+	public kyber.Point
+}
+
+// NewOrganizerHub returns a Organizer Hub.
+func NewWitnessHub(public kyber.Point) Hub {
+	return &witnessHub{
+		messageChan: make(chan IncomingMessage),
+		channelByID: make(map[string]Channel),
+		public:      public,
+	}
+}
+
+func (w *witnessHub) RemoveClientSocket(client *ClientSocket) {
+	//TODO
+}
+
+func (w *witnessHub) Recv(msg IncomingMessage) {
+	log.Printf("witnessHub::Recv")
+	w.messageChan <- msg
+}
```

Comment:
Looks good, but it's unclear to me who and where would call this function?

Reply:
I believe it is called by any socket (witness, organizer or client) to which the witness is connected in the ReadPump() function.